### PR TITLE
[Fix]: Searchbar doesn't lose focus when tapped outside

### LINKED ISSUE #145096: Expose onTapOutside from SearchBar's TextField
### Use case

All the `TextField`s in my app are set to lose focus closing the keyboard when they're tapped outside. I can't do the same on `SearchBar` as the `onTapOutside` property of `TextField` class is not exposed.

### Proposal

Introduce `onTextOutside` parameter for `SearchBar` and `SearchAnchor` constructors.

### PR DESCRIPTION
**Pull request by Sahil-kachhap**
Added a Fix to search bar that allows it be unfocused when tapped anywhere outside the search bar.

I have attached below the app flow after implementing the fix.

https://github.com/flutter/flutter/assets/54017876/70915c47-9b77-4a43-a128-8706107f921f

Issue that gets resolved by this fix: #145096

*If you had to change anything in the [flutter/tests] repo, include a link to the migration guide as per the [breaking change policy].*

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [ ] I followed the [breaking change policy] and added [Data Driven Fixes] where supported.
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat
[Data Driven Fixes]: https://github.com/flutter/flutter/wiki/Data-driven-Fixes

### CODE REVIEW

**Comment by Sahil-kachhap** [comment]
@HansMuller can you please assign a reviewer to this PR

File: packages/flutter/test/material/search_anchor_test.dart [hunk 1]
```diff
@@ -3061,6 +3061,51 @@ void main() {
...
+        MaterialApp(
+          home: Scaffold(
+            body: SearchAnchor(
+              builder: (BuildContext context, SearchController controller){
+                return SearchBar(
+                  controller: controller,
+                  onTap: () {
+                    controller.openView();
+                  },
+<<<<<<< HEAD
```
Comment (gspencergoog) [inline packages/flutter/test/material/search_anchor_test.dart:3077]: Looks like you need to resolve these conflicts.

**Review by gspencergoog — APPROVED** [review]
32384589-a60f0e74-c078-11e7-9bc1-e5b5287aea9d

(once the conflict markers are removed)

**Comment by Sahil-kachhap** [comment]
> 32384589-a60f0e74-c078-11e7-9bc1-e5b5287aea9d
> 
> (once the conflict markers are removed)

Fixed the conflict.

**Comment by Sahil-kachhap** [comment]
@gspencergoog, @HansMuller I have fixed the minor problems that lead to test failures. Now all tests have passed. Will this be auto merged ?

**Review by HansMuller — APPROVED** [review]
LGTM

**Comment by mattalbr** [comment]
Thanks for this PR! Maybe a silly question, but is there a reason unfocusing on tap outside isn't the default? I looked through the material guidelines and couldn't find anything explicitly called out around this behavior, but Gmail and YouTube (who I'm assuming are up-to-date on Material) both unfocus on tap outside

**Pull request merged**